docs/docs/typescript/strict-mode.md

```diff
@@ -0,0 +1,239 @@
+---
+description: TS Strict mode tips and tricks
+---
+
+# TypeScript Strict Mode
+
+Looks like you're ready to level up your TypeScript game!
+Redwood supports [strict mode](https://www.typescriptlang.org/docs/handbook/2/basic-types.html#strictness), but doesn't enable it by default.
+While strict mode gives you a lot more safety, it makes your code a bit more verbose and requires you to make small manual changes if you use the generators.
+
+## Enabling strict mode
+
+Enable strict mode by setting `strict` to true in `web/tsconfig.json` and `api/tsconfig.json`:
+
+```json title="web/tsconfig.json, api/tsconfig.json"
+{
+  "compilerOptions": {
+    "noEmit": true,
+    "allowJs": true,
+    // highlight-next-line
+    "strict": true,
+    // ...
+  }
+  // ...
+}
+```
+
+Redwood's type generator behaves a bit differently in strict mode, so now that you've opted in, make sure to generate types:
+
+```
+yarn rw g types
+```
+
+## Manual tweaks to generated code
+
+Now that you're in strict mode, there are some changes you need to make to get rid of those pesky red underlines!
+
+### Service functions & tests
+
+By default, Redwood's generators assume that Service functions don't require any parameters. For example, [in the tutorial](../tutorial/chapter2/side-quest.md), the `posts` Service function generated by the scaffold command doesn't specify any parameters:
+
+```ts title="src/services/posts/posts.test.ts"
+// highlight-next-line
+export const posts: QueryResolvers['posts'] = () => {
+  return db.post.findMany()
+}
+```
+
+While this is true, in the strictest sense, all GraphQL resolvers (and most of your Service functions _do_ end up becoming GraphQL resolvers) take some sort of input, so you'll need to modify the calls to your Service functions that have been typed with `QueryResolvers` or `MutationResolvers`:
+
+```ts title="src/services/posts/posts.test.ts"
+describe('posts', () => {
+  scenario('returns all posts', async (scenario: StandardScenario) => {
+    const result = await posts() // 🛑 error
+    const result = await posts({}) // ✅
+
+    expect(result.length).toEqual(Object.keys(scenario.post).length)
+  })
+})
+```
+
+:::info Why this happens in strict mode
+
+In strict mode, Redwood generates different types for `QueryResolvers` and `MutationResolvers`, where the first argument—`args`—is no longer optional.
```

Comment:
This used to be "args is no longer non-optional". That was too many mental cycles for me, but did I get it right as "no longer optional"? Or is it required?

Reply:
Whoops 😆, yeah this is correct. Args is required.